### DISCUSSION TITLE
Retrieve ItemsPanel of ListBox in code behind

### DISCUSSION THREAD

**Discussion by agnauck** [AvaloniaUI/Avalonia#10426] [General]
I have a virtualized ListBox like this
```
 <ListBox Name="MyList">
        <ListBox.ItemsPanel>
            <ItemsPanelTemplate>
                <VirtualizingStackPanel Name="VirtualizingStackPanel" Orientation="Vertical"/>
            </ItemsPanelTemplate>
        </ListBox.ItemsPanel>
    </ListBox>
```

how can I retrieve the VirtualizingStackPanel in code behind?

I need the VirtualizingStackPanel to find out which items are currently displayed. I am planning to persist the last visible item and scroll to it when the view is getting restored later in the app.

Or is there some better way of doing this?
I cannot rely on scroll position because items count could have increased when the view is getting restored.

Thanks,
Alex

**Comment by grokys** [comment]
If you're using 0.10.x you can use `MyList.Presenter.Panel` to get the panel. In 11.0-preview4 and later we've also added `ItemsPanelRoot` for compatibilty with WPF.

**Comment by agnauck** [comment]
Thanks, its working now.
I tried to get the Panel always in the ListBox `OnTemplateApplied`  event which is too early.
